### PR TITLE
osd metadata additions

### PR DESCRIPTION
**Pull request by joehandzik**
Hey guys,

Here's my set of changes to propagate the device node stuff I added. Tested a bunch over the last day with minion-sim, so I'm pretty confident in everything outside of the ceph.py salt code. Feedback across the board would be appreciated. For reference, here's my Ceph commit that goes with all this: https://github.com/ceph/ceph/commit/4fe62cd7a1761c73ab526978a0a6a3cab4f8e2c4

It's amazing what you can get done when you use Salt 2014. :P

### CODE REVIEW

File: rest-api/calamari_rest/serializers/v2.py [hunk 1]
```diff
@@ -142,6 +142,8 @@ class Meta:
     public_addr = serializers.CharField(read_only=True, help_text="Public/frontend IP address")
     cluster_addr = serializers.CharField(read_only=True, help_text="Cluster/backend IP address")
     crush_node_ancestry = serializers.Field(help_text="An ordered list of CRUSH node ids that represent a path from the parent node of this OSD up to the root of the tree")
+    backend_partition_path = serializers.CharField(read_only=True, help_text="Full path to the storage partition targeted by this OSD")
```
Comment (GregMeno) [inline rest-api/calamari_rest/serializers/v2.py]: @joehandzik What provision do you make for cases where an OSD has it's journal and data on different devices?

Comment (joehandzik) [inline rest-api/calamari_rest/serializers/v2.py]: None. This set of changes is only to address information about data partitions and devices (hence the "backend" modifier prepended to both new elements, leaving room for a "journal" version). We can make a set of changes around journals if we decide it's appropriate (which, at some point, I expect to do). It'll just involve a call to the function I created to pull the data device information, but targeted at the journal device instead. I'd rather focus on creating the entire set of information and capabilities for the data devices first, personally.

Comment (GregMeno) [inline rest-api/calamari_rest/serializers/v2.py]: @joehandzik a separate change-set sounds appropriate to me

Comment (joehandzik) [inline rest-api/calamari_rest/serializers/v2.py]: Ok, good deal. I'll get this tested against a version of Ceph that has my required changes as soon as I can, and I'll give you a thumbs up or a thumbs down for a integration then. Sound fair? Until then, if there's anything you want me to change stylistically, or any other tests I should be adding, feel free to let me know.

**Comment by GregMeno** [comment]
@joehandzik have you had a chance to run the integration tests?

**Comment by joehandzik** [comment]
Nope, between some personal stuff and a set of other work I'm pulled off on temporarily I have not. Hoping to get to it this week. I'll keep you posted!

**Comment by GregMeno** [comment]
Ran 11 tests in 17.234s

FAILED (errors=1)
(env)vagrant@vagrant-ubuntu-trusty-64:~/calamari$ nosetests -v -x tests/test_osd_management.py
That we can apply ceph commands to an OSD ... ERROR

======================================================================
ERROR: That we can apply ceph commands to an OSD
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/calamari/tests/test_osd_management.py", line 76, in test_apply
    fsid = self._wait_for_cluster()
  File "/home/vagrant/calamari/tests/server_testcase.py", line 88, in _wait_for_cluster
    scalable_wait_until_true(lambda: self._maps_populated(cluster_id))
  File "/home/vagrant/calamari/tests/utils.py", line 43, in scalable_wait_until_true
    return wait_until_true(condition, timeout)
  File "/home/vagrant/calamari/tests/utils.py", line 30, in wait_until_true
    raise WaitTimeout("After %s seconds (at %s)" % (elapsed, datetime.datetime.utcnow().isoformat()))
WaitTimeout: After 10 seconds (at 2015-08-11T19:16:31.883310)
-------------------- >> begin captured stdout << ---------------------
404: "Object of type osd_map with id ['metadata_by_id'] not found"

**Comment by GregMeno** [comment]
@joehandzik I took a shot at fixing it here:
https://github.com/ceph/calamari/commit/248624cef0e094ea2c20ad7ed3735e86e861fd69
there is a super easy way to get tests running using wip-converge. Let's talk about that when you get back. It seems like this branch is mostly working.

**Comment by joehandzik** [comment]
Ah, it was that simple. Like I said a couple weeks ago, the only issue I was experiencing was...actually calling osd metadata. Was there an intelligent way for me to know that I needed to add the id to the argdict argument, instead of trying to use the 'osd metadata' command like I was using it via the command line? Aside from reading all the code associated in the transaction, of course. :)

Also, possibly a silly question but...is there a particular reason why you changed the data.update(updated_osd_metadata) to a data.append(updated_osd_metadata)? It seems trivial to me, and I don't prefer one or the other, but it'd be great to know if there was a particular reason why you did that.

**Comment by GregMeno** [comment]
@joehandzik welcome back! there is no way to know beyond reading the code https://github.com/ceph/ceph/blob/master/src/mon/MonCommands.h#L437 AFAIK

as for update vs append:
data['osd_metadata'] was a list of dicts in the test json which is what the cthulhu code seemed to expect.
so I changed it to a list and lists don't have update()
Does that make sense?

**Comment by joehandzik** [comment]
Fair enough, and makes perfect sense. Thanks!

**Comment by dmick** [comment]
ceph --verbose will show you the json sent for a command

**Comment by GregMeno** [comment]
@joehandzik Thank you for the contribution!

**Comment by joehandzik** [comment]
@GregMeno You're welcome! Thanks for the help getting it finished off while I was out.

**Pull request merged**